### PR TITLE
Use as_search if it's available, even if our index mapping is not an

### PR DESCRIPTION
**Pull request by dcromer**
object.

### CODE REVIEW

File: lib/elastic_record/callbacks.rb [hunk 1]
```diff
@@ -60,7 +60,11 @@ def elastic_search_value(field, mapping)
         when :nested
           value.map(&:as_search)
         else
```
Comment (matthuhiggins) [inline lib/elastic_record/callbacks.rb]: There ought to be a case statement for `: geo_point`.

Comment (dcromer) [inline lib/elastic_record/callbacks.rb]: So, your attribute definition looks like:

```
  "manual_geocoordinate": {
    "type": "geo_point",
    "class_name": "GeoPoint",
    "attributes": {
      "lat": {
        "type": "float"
      },
      "lon": {
        "type": "float"
      }
    }
  },
```

And then [axle_attributes](https://github.com/data-axle/axle_attributes/blob/4c130518b993d40c6de7ea79dfa5b988e7ba80d6/lib/axle_attributes/definition.rb) would translate that into elastic_record/superstore-speak?

Comment (matthuhiggins) [inline lib/elastic_record/callbacks.rb]: I am starting to like your geopoint type that is native in superstore.

**Comment by ebarendt** [comment]
Is this still needed?

**Pull request closed without merging**